homeassistant/components/econet/sensor.py

```diff
@@ -1,103 +1,158 @@
+        name="compressor_health",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=None,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    EcoNetSensorEntityDescription(
+        key="override_status",
+        name="override_status",
+        native_unit_of_measurement=None,
+        device_class=None,
+        state_class=None,
+    ),
+    EcoNetSensorEntityDescription(
+        key="todays_water_usage",
+        name="water_usage_today",
+        native_unit_of_measurement=UnitOfVolume.GALLONS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    EcoNetSensorEntityDescription(
+        key="todays_energy_usage",
+        name="power_usage_today",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    EcoNetSensorEntityDescription(
+        key="alert_count",
+        name="alert_count",
+        native_unit_of_measurement=None,
+        device_class=None,
+        state_class=None,
+    ),
+    EcoNetSensorEntityDescription(
+        key="wifi_signal",
+        name="wifi_signal",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    EcoNetSensorEntityDescription(
+        key="running_state",
+        name="running_state",
+        native_unit_of_measurement=None,
+        device_class=None,
+        state_class=None,
+    ),
+)
+
+
+ENERGY_KILO_BRITISH_THERMAL_UNIT = "kBtu"
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up EcoNet sensor based on a config entry."""
 
-    equipment = hass.data[DOMAIN][EQUIPMENT][entry.entry_id]
-    sensors = []
-    all_equipment = equipment[EquipmentType.WATER_HEATER].copy()
-    all_equipment.extend(equipment[EquipmentType.THERMOSTAT].copy())
+    data = hass.data[DOMAIN][EQUIPMENT][entry.entry_id]
+    equipment = data[EquipmentType.WATER_HEATER].copy()
+    equipment.extend(data[EquipmentType.THERMOSTAT].copy())
 
-    for _equip in all_equipment:
-        for name, attribute in SENSOR_NAMES_TO_ATTRIBUTES.items():
-            if getattr(_equip, attribute, None) is not None:
-                sensors.append(EcoNetSensor(_equip, name))
-        # This is None to start with and all device have it
-        sensors.append(EcoNetSensor(_equip, WIFI_SIGNAL))
+    sensors = []
 
-    for water_heater in equipment[EquipmentType.WATER_HEATER]:
-        # These aren't part of the device and start off as None in pyeconet so always add them
-        sensors.append(EcoNetSensor(water_heater, WATER_USAGE_TODAY))
-        sensors.append(EcoNetSensor(water_heater, POWER_USAGE_TODAY))
+    sensors = [
+        EcoNetSensor(_equip, description.name, description)
```

Comment:
No need to pass the name anymore:
```suggestion
        EcoNetSensor(_equip, description)
```